packages/core-flows/src/definition/cart/workflows/refresh-payment-collection.ts

```diff
@@ -0,0 +1,75 @@
+import {
+  StepResponse,
+  WorkflowData,
+  createStep,
+  createWorkflow,
+  transform,
+} from "@medusajs/workflows-sdk"
+import { useRemoteQueryStep } from "../../../common/steps/use-remote-query"
+import {
+  deletePaymentSessionStep,
+  updatePaymentCollectionStep,
+} from "../../payment-collection"
+
+type WorklowInput = {
+  cart_id: string
+}
+
+interface StepInput {
+  cart_id: string
+}
+
+// We export a step running the workflow too, so that we can use it as a subworkflow e.g. in the update cart workflows
+export const refreshPaymentCollectionForCartStepId =
+  "refresh-payment-collection-for-cart"
+export const refreshPaymentCollectionForCartStep = createStep(
+  refreshPaymentCollectionForCartStepId,
+  async (data: StepInput, { container }) => {
+    await refreshPaymentCollectionForCartWorkflow(container).run({
```

Comment:
discussion: As I was explaining, using workflows in workflows can be useful, but there is no easy way to no that a workflow is being used in another workflow and that can bring some downside. As usual, responsibility principle and domain concerns must be taken into consideration and updating a workflow should necessarely impact the workflows that are using another workflow as the changes might not be related but still will impact the other workflows. So I just want to put a warning on those usage to be more cautious in those context

Reply:
We will likely need to address this more generally for 2.0 to have a pattern in place.